packages/extension-ui/src/Popup/Signing/Request.tsx

```diff
@@ -3,37 +3,43 @@
 // of the Apache-2.0 license. See the LICENSE file for details.
 
 import { ExtrinsicPayload } from '@polkadot/types/interfaces';
-import { AccountJson, RequestExtrinsicSign } from '@polkadot/extension/background/types';
+import { AccountJson, RequestSign } from '@polkadot/extension/background/types';
 
 import React, { useContext, useState, useEffect } from 'react';
 import { createType } from '@polkadot/types';
 
 import { ActionBar, ActionContext, Address, Link } from '../../components';
 import { approveSignPassword, approveSignSignature, cancelSignRequest } from '../../messaging';
-import Details from './Details';
+import Extrinsic from './Extrinsic';
+import Bytes from './Bytes';
 import Qr from './Qr';
 import Unlock from './Unlock';
+import { SignerPayloadJSON, SignerPayloadRaw } from '@polkadot/types/types';
 
 interface Props {
   account: AccountJson;
   isFirst: boolean;
-  request: RequestExtrinsicSign;
+  request: RequestSign;
   signId: string;
   url: string;
 }
 
 export default function Request ({ account: { isExternal }, isFirst, request, signId, url }: Props): React.ReactElement<Props> | null {
   const onAction = useContext(ActionContext);
-  const [payload, setPayload] = useState<ExtrinsicPayload | null>(null);
+  const [hexBytes, setHexBytes] = useState<string | null>(null);
+  const [extrinsic, setExtrinsic] = useState<ExtrinsicPayload | null>(null);
 
   useEffect((): void => {
-    setPayload(createType('ExtrinsicPayload', request, { version: request.version }));
+    const inner = request.getInner();
```

Comment:
Probably would have implemented this as an enum-like structure, where you can do `isRaw` & `getRaw` (and `isPayload` & `getPayload`, but this is fine, it works as expected.